### PR TITLE
#22 Update schemas

### LINKED ISSUE #22: Update schemas
There is update on schemas

### PR DESCRIPTION
**Pull request by ABMC831**
Updates schema to ADR 009 as of 2024-10-17
closes #22

### CODE REVIEW

File: conf/topics.json [hunk 1]
```diff
@@ -1,122 +1,119 @@
...
-            "guid",
-            "app_id_snow",
-            "source_app",
-            "timestamp_start",
-            "timestamp_end",
-            "data_definition_id",
-            "status"
-        ],
-        "type": "object"
+        "required": ["event_id", "job_ref", "tenant_id", "source_app", "source_app_version", "environment", "timestamp_start", "timestamp_end", "jobs", "jobs.catalog_id", "jobs.status", " jobs.timestamp_start", "jobs.timestamp_end"]
```
Comment (yruslan) [inline conf/topics.json:66]: Not 100% sure how it works when `jobs` is a mandatory field. It is an array, and it is possible to have an empty array when the pipeline has failed and was not able to run any jobs. For instance, when EMR is not available 😆

File: scripts/notebook.ipynb [hunk 2]
```diff
@@ -84,17 +84,18 @@
...
-    "        \"schema_link\": \"https://schema-registry.company.com/schemas/data-def-5678\",\n",
-    "        \"source_app\": \"DataIngestionApp\",\n",
-    "        \"timestamp_event\": 1657896543210\n",
-    "    })\n",
+    "        \"event_id\": \"JupyterEventId\",\n",
+    "        \"tenant_id\": \"JupyterTenantId\",\n",
+    "        \"source_app\": \"JupyterSrc\",\n",
+    "        \"source_app_version\": \"v2024-10-17\",\n",
+    "        \"environment\": \"JupyterEnv\",\n",
+    "        \"timestamp_event\": 20241017,\n",
```
Comment (yruslan) [inline scripts/notebook.ipynb:92]: Just to stress this is is epoch milli
```suggestion
    "        \"timestamp_event\": 1729602770000,\n",
```

**Review by yruslan — APPROVED** [review]
Looks good. Just one thing to check when we start using it. It should be allowed to send a run event with the empty jobs array to the run topic... i think.

File: scripts/notebook.ipynb (hunk 2 above)
Comment (ABMC831) [reply scripts/notebook.ipynb]: I mean, that is sample script, might as well be there 1234 for easy visual check message arrived... but sure why not

File: conf/topics.json (hunk 1 above)
Comment (ABMC831) [reply conf/topics.json:66]: actually.... we need to remove
jobs.*   required fields... as jsonschema validate complains as that prescribes them to be present in underlying object, not in the element of a array

those are
jobs.catalog_id
jobs.status
jobs.timestamp_start
jobs.timestamp_end
as none of those would be present at that xpath

as for jobs..... schema just requires field to be there and to be array.... so empty array is fine.... null is not fine though

**Pull request merged**